lib/kimono_labs_client/web_client.rb

```diff
@@ -0,0 +1,42 @@
+module KimonoLabsClient
+  # Web client for fetching a json endpoint's content via http.
+  class WebClient
+    attr_reader :api_key
+
+    # initializer
+    def initialize(api_key)
+      @api_key = api_key
+      @http = http
+
+      fail 'API key must be provided!' unless key?
+    end
+
+    def key?
```

Comment:
This is a good refactoring but perhaps we can make it better? Before I offer specific examples I'd encourage a 30 minute break to watch Avdi Grimm's talk on the [confident ruby](http://www.confidentruby.com/) page. There may be some good approaches there. Also, I should really buy his book already.